lib/reporter/index.js

```diff
@@ -180,13 +180,17 @@ class Reporter extends SimplifiedReporter {
     try {
       let commandResult = result;
 
-      const isSuccess =  result === undefined ||
+      let isSuccess =  result === undefined ||
         result == null ||
         !((result instanceof Error) ||
           (result.error instanceof Error) ||
           result.status === -1
         );
 
+      if (node.fullName === 'waitForElementNotPresent') {
+        isSuccess = !(result.name  === 'NightwatchAssertError');
+      }
+
```

Comment:
Shouldn't we do this every node and add a condition to verify it is an assert type node?

Reply:
For every other node the status indicates the correct outcome. Only for `waitForElementNotPresent` the status codes are reversed.

Reply:
@gravityvi Can there be other assert nodes with wrong status codes? Ideally all results should have consistent status codes, but cannot change this for backward compatibility.

Reply:
Why can't we add this condtion `result.name === NightwatchAssertError` for every node?

Reply:
I guess we can, can't think of any case where this results in success. Will update.

Reply:
For all other nodes, the definition of success is captured accurately by the `isSuccess` condition:
```js
let isSuccess =  result === undefined ||
        result == null ||
        !((result instanceof Error) ||
          (result.error instanceof Error) ||
          result.status === -1
        );
```

In case of `waitForElementNotPresent`, the success case also contains error. To reset the success case I am using `result.name === NightwatchAssertError`. If we use this for all nodes then that will cause other nodes which are not asserts (other cases might exist) to show wrong success value.